### ISSUE TITLE
Angular looks for the template of a service

### ISSUE THREAD

**Issue by zoechi** [dart-lang/angular2#351]
```dart
import 'package:angular2/angular2.dart' show Injectable;
import 'package:fhir_webui/webui.dart' show EditMode;

@Injectable()
class EditModeService {
  EditMode _mode = EditMode.view;

  EditMode get mode => _mode;
  set mode(EditMode value) => _mode = value ?? EditMode.view;
}
```
```
[web] GET packages/fhir_webui/src/webui_generated/form/edit_mode_service.template.dart => Could not find asset fhir_webui|lib/src/webui_generated/form/edit_mode_service.template.dart.
```

I get this error even when the service isn't used anywhere within the application (only used by another application that also depends on the package that contains it)

When I comment out `@Injectable()` the error goes away, but of course breaks my application in other ways.

**Comment by matanlurey** [comment]
`.template.dart` is a bit of a misnomer, it can also mean "reflection information".

Could you share where `edit_mode_service.dart` is imported?

**Comment by zoechi** [comment]
`fhir_webui/lib/webui.dart`

```dart
export 'package:fhir_webui/src/webui_generated/form/edit_mode_service.dart';
```
`demo2/lib/generated/components/form_account_component.dart`
```dart
import 'package:fhir_webui/src/webui_generated/form/edit_mode_service.dart';
...
@Component(
    selector: 'form-ce1e8325-e814-49f2-ad0f-fb10f21604a3',
    templateUrl: 'form_account_component.html',
    styleUrls: const [
      'form_account_component.css'
    ],
    providers: const <dynamic>[
      ToolbarCrudButtonsService,
      EditModeService,
```

**Comment by zoechi** [comment]
I just changed it to import the file containing the class directly (without the `webui.dart` file that exports the file)

Now I get

```
[web] GET packages/fhir_webui/webui/toolbar/toolbar_button_component.template.dart => Could not find asset fhir_webui|lib/webui/toolbar/toolbar_button_component.template.dart.
[web] GET packages/fhir_webui/src/webui_generated/primitive_value_input/xhtml_input_component.template.dart => Could not find asset fhir_webui|lib/src/webui_generated/primitive_value_input/xhtml_input_component.template.dart.
```

It doesn't look like a real problem in my code. Commenting something in/out (like `@Injectable()` mentioned in the initial post) while `pub serve` is running makes the problem go away temporarily.

**Comment by zoechi** [comment]
Deleted my previous comment, this was caused of a broken import, because I commented out above export to import the file directly.

**Comment by matanlurey** [comment]
Do we still need a resolution? Confused.

**Comment by zoechi** [comment]
My last comment was only related to my 2nd-last comment, which I deleted.

Changing the import to import the service directly, instead of via a file that exports it, seems to have fixed it for now. But I never know how long this keeps that fixed. 
Random changes fixed it before, but usually not for long.
Is it supposed to work with export/import?

**Comment by matanlurey** [comment]
I'm not sure export/import works properly with pub transformers right now to be honest.

**Comment by zoechi** [comment]
https://github.com/dart-lang/angular2_components/blob/master/lib/angular2_components.dart does it even 2 levels

**Comment by zoechi** [comment]
When at least the error message would provide anything useful what causes the issue.
The only information it provides is that it's related somehow to a class,
and that it is so random makes it even more confusing and annoying.

**Comment by zoechi** [comment]
With `pub build` now it complains about `UndoService`, and if I comment `@Injectable()` out it complains about `GlobalsService`. 
I also changed all imports to import these services directly (without an intermediate file that exports them) but that didn't change anything this time.

I think somehow it just randomly complains about "something", even though there's probably nothing wrong with my code, because when I overcome the error somehow, the application is working fine.

I hope the new code generation becomes available soon and fixes this issue. This is really unbearable.

**Comment by zoechi** [comment]
After being able to work around #348 with the help of @leonsenft, 
I was able to "fix" this one as well.

After getting rid of the CSS errors from 348, I got some errors that I didn't get before,
for hacks that were previously required to work around this issue 
(but actually the hacks should have caused the errors before because of the hacks). 
After fixing these errors the application started working again.

So, now it's working without any weird hacks. I didn't have this since months.
